crypto-ffi/bindings/jvm/src/main/kotlin/com/wire/crypto/CoreCrypto.kt

```diff
@@ -16,22 +16,22 @@ suspend operator fun CoreCrypto.Companion.invoke(
     databaseKey: DatabaseKey
 ) = coreCryptoDeferredInit(keystore, databaseKey, null).lift()
 
-/**
- * Instantiate a history client.
- *
- * This client exposes the full interface of `CoreCrypto`, but it should only be used to decrypt messages.
- * Other use is a logic error.
- */
-suspend fun historyClient(historySecret: HistorySecret) = coreCryptoHistoryClient(historySecret).lift()
-
 /**
  * A high-level wrapper around a CoreCrypto client as emitted by Uniffi.
  *
  * This wrapper should be largely transparent to end users. It exists to improve the
  * callback interfaces: `.transaction(...)`, `.registerFooObserver(...)`, etc.
  */
 class CoreCryptoClient(private val cc: CoreCrypto) {
-    companion object
+    companion object {
+        /**
+         * Instantiate a history client.
+         *
+         * This client exposes the full interface of `CoreCrypto`, but it should only be used to decrypt messages.
+         * Other use is a logic error.
+         */
+        suspend fun historyClient(historySecret: HistorySecret) = coreCryptoHistoryClient(historySecret).lift()
```

Comment:
So to construct a history client, you need to call `CoreCryptoClient.historyClient`. This is the only public constructor defined on `CoreCryptoClient`. 

I'd prefer that it were different, but I suspect that if the FFI class gets renamed then my objection may become obsolete.